packages/components/addon/components/hds/code-block/index.js

```diff
@@ -0,0 +1,151 @@
+
+// these are picked up from /public/ – https://cli.emberjs.com/release/writing-addons/#public
+Prism.plugins.autoloader.languages_path =
+  '/@hashicorp/design-system-components/prism-languages/';
+
+export default class HdsCodeBlockIndexComponent extends Component {
+  @tracked prismCode = '';
+
+  /**
+   * Generates a unique ID for the code content
+   *
+   * @param preCodeId
+   */
+  preCodeId = 'pre-code-' + guidFor(this);
+
+  /**
+   * @param code
+   * @type {string}
+   * @description code text content for the CodeBlock
+   */
+  get code() {
+    const code = this.args.value;
+
+    assert(
+      '@code for "Hds::CodeBlock" must have a valid value',
+      code !== undefined
+    );
+
+    if (Prism?.plugins?.NormalizeWhitespace) {
+      return Prism.plugins.NormalizeWhitespace.normalize(code);
+    }
+
+    return code;
+  }
+
+  /**
+   * @param language
+   * @type {string}
+   * @default undefined
+   * @description name of coding language used within CodeBlock for syntax highlighting
+   */
+  get language() {
+    return this.args.language ?? undefined;
+  }
+
+  get languageClass() {
+    return `language-${this.language}`;
+  }
+
+  /**
+   * @param hasLineNumbers
+   * @type {boolean}
+   * @default true
+   * @description Displays line numbers if true
+   */
+  get hasLineNumbers() {
+    return this.args.hasLineNumbers ?? true;
+  }
+
+  /**
+   * @param isReadOnly
+   * @type {boolean}
+   * @default true
+   * @description Make text content non-editable / editable
+   */
+  get isReadOnly() {
+    return this.args.isReadOnly ?? true;
+  }
+
+  /**
+   * @param isRounded
+   * @type {boolean}
+   * @default true
+   * @description Make CodeBlock container corners appear rounded
+   */
+  get isRounded() {
+    return this.args.isRounded ?? true;
+  }
+
+  @action
+  setPrismCode(element) {
+    const code = this.code;
+    const language = this.language;
+    const grammar = Prism.languages[language];
+
+    if (code && language && grammar) {
+      this.prismCode = htmlSafe(Prism.highlight(code, grammar, language));
+    } else {
+      this.prismCode = htmlSafe(code);
+    }
+
+    // Force plugin initialization, required for Prism.highlight usage.
+    // See https://github.com/PrismJS/prism/issues/1234
+    Prism.hooks.run('complete', {
+      code,
+      element,
+    });
+  }
+
+  /**
+   * Get the class names to apply to the component.
+   * @method classNames
+   * @return {string} The "class" attribute to apply to the component.
+   */
+  get classNames() {
+    // Currently theres is only one theme so the class name is hard-coded.
+    // In the future, additional themes such as a "light" theme could be added.
+    let classes = ['hds-code-block', 'hds-code-block--theme-dark'];
+
+    if (this.language) {
+      classes.push(`language-${this.language}`);
+    }
+
+    if (this.args.hasCopyButton) {
+      classes.push('hds-code-block--has-copy-button');
+    }
```

Comment:
As mentioned in a comment in CSS, this can be removed and the position absolute assigned directly to the copy button element.
```suggestion
```

Reply:
Removed in 97f0f1ef9231f5b11f73f7c39074eb9b1cbcec6c when refactoring the CopyButton position